Controller/Annotations/AbstractParam.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the FOSRestBundle package.
+ *
+ * (c) FriendsOfSymfony <http://friendsofsymfony.github.com/>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace FOS\RestBundle\Controller\Annotations;
+
+use Symfony\Component\Validator\Constraints;
+use Symfony\Component\DependencyInjection\ContainerAware;
+
+/**
+ * {@inheritdoc}
+ *
+ * @author Jordi Boggiano <j.boggiano@seld.be>
+ * @author Boris Guéry <guery.b@gmail.com>
+ * @author Ener-Getick <egetick@gmail.com>
+ */
+abstract class AbstractParam extends ContainerAware implements ParamInterface
```

Comment:
The ``Container`` is used to resolve the parameter in the default value. We should maybe move this to the ``ParamFetcher`` ...